Assets/HoloToolkit/UX/Scripts/Progress/LoadingAnimation.cs

```diff
@@ -0,0 +1,237 @@
+        private bool mLoopPause = false;
+        private int mFadeIndex = 0;
+        private bool mCheckLoopPause = false;
+        private Vector3 mPositionVector;
+        private Vector3 mRotatedPositionVector;
+        private LoadingAnimation loadingAnimation;
+
+        public GameObject[] Orbs
+        {
+            get
+            {
+                return orbs;
+            }
+
+            set
+            {
+                orbs = value;
+            }
+        }
+
+        private void Start()
+        {
+            mPositionVector = transform.up;
+
+            if (!Mathf.Approximately(Vector3.Angle(Axis, mPositionVector), 90))
+            {
+                mPositionVector = transform.forward;
+                if (!Mathf.Approximately(Vector3.Angle(Axis, mPositionVector), 90))
+                {
+                    float x = Mathf.Abs(Axis.x);
+                    float y = Mathf.Abs(Axis.y);
+                    float z = Mathf.Abs(Axis.z);
+
+                    if (x > y && x > z)
+                    {
+                        // left or right - cross with the z axis
+                        mPositionVector = Vector3.Cross(Axis * Radius, Vector3.forward);
+                    }
+
+                    if (z > y && z > x)
+                    {
+                        // forward or backward - cross with the x axis
+                        mPositionVector = Vector3.Cross(Axis * Radius, Vector3.right);
+                    }
+
+                    if (y > z && y > x)
+                    {
+                        // up or down - cross with the x axis
+                        mPositionVector = Vector3.Cross(Axis * Radius, Vector3.right);
+                    }
+                }
+            }
+        }
+
+        public void StartLoader()
+        {
+            loadingAnimation = gameObject.GetComponent<LoadingAnimation>();
+            for (int i = 0; i < Orbs.Length; ++i)
+            {
+                Orbs[i].SetActive(false);
+                FadeObject fade = Orbs[i].GetComponent<FadeObject>();
+                fade.ResetFade(0);
+            }
+
+            mStartingLoader = true;
+            mStartingIndex = 0;
+            mRevolutionsCount = 0;
+            IsPaused = false;
+        }
+
+        public void StopLoader()
+        {
+            for (int i = 0; i < Orbs.Length; ++i)
+            {
+                Orbs[i].gameObject.SetActive(false);
+            }
+        }
+
+        public void StartOrbit()
+        {
+            IsPaused = false;
+        }
+
+        public void ResetOrbit()
+        {
+            mAngle = 0;
+        }
+
+        public float QuartEaseInOut(float s, float e, float v)
+        {
+            //e -= s;
+            if ((v /= 0.5f) < 1)
+                return e / 2 * v * v * v * v + s;
+
+            return -e / 2 * ((v -= 2) * v * v * v - 2) + s;
+        }
+
+        private void Update()
+        {
+            if (IsPaused) return;
```

Comment:
event though this is a simple one line if, please wrap the return statement in { }